From 037ece89c3c7c1daa0220166c9652fdca4a28c32 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 27 Aug 2024 00:51:17 +0530
Subject: [PATCH] Fix for flickering issue

In Android 15, variable refresh rate(vrr) is enabled by default.
As the end to end variable refresh rate support is not yet enabled,
flickering issue seen during refresh rate change.

Fix the issue by disabling the vrr and frame rate override.

Tests done:
- Android boot in CIV Gvt-d and BM config
- adb reboot
- Wi-Fi on/off/scan/connect/disconnect with static ip config
- Browsing

Tracked-On: OAM-123919
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 groups/graphics/auto/init.rc    | 2 ++
 groups/graphics/auto/product.mk | 1 +
 2 files changed, 3 insertions(+)

diff --git a/groups/graphics/auto/init.rc b/groups/graphics/auto/init.rc
index 871e573..a155ec4 100755
--- a/groups/graphics/auto/init.rc
+++ b/groups/graphics/auto/init.rc
@@ -5,6 +5,8 @@ on init
 
 on post-fs-data
    #setprop debug.sf.nobootanimation 1
+   setprop debug.sf.enable_vrr_config false
+   setprop debug.sf.vrr_timeout_hint_enabled false
    mkdir /data/system 0770 system system
 
 on boot
diff --git a/groups/graphics/auto/product.mk b/groups/graphics/auto/product.mk
index 45d033b..b998257 100755
--- a/groups/graphics/auto/product.mk
+++ b/groups/graphics/auto/product.mk
@@ -42,6 +42,7 @@ PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.max_frame_buffer_acquir
 # System's VSYNC phase offsets in nanoseconds
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.vsync_event_phase_offset_ns=7500000
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.vsync_sf_event_phase_offset_ns=3000000
+PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.surface_flinger.enable_frame_rate_override=false
 
 # i915 firmwares
 $(foreach fw,$(I915_FW),$(eval PRODUCT_PACKAGES += $(notdir $(fw))))
-- 
2.45.2

